ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -357,32 +385,40 @@ public void replicaTokenTest() throws InterruptedException {
    */
   private void addDeleteMessagesToReplicasOfPartition(PartitionId partitionId, StoreKey id, List<Host> hosts)
       throws MessageFormatException, IOException {
-    ByteBuffer buffer = getDeleteMessage(id);
+    MessageInfo putMsg = getMessageInfo(id, hosts.get(0).infosByPartition.get(partitionId), false);
+    long deletionTimeMs = System.currentTimeMillis();
+    ByteBuffer buffer = getDeleteMessage(id, putMsg.getAccountId(), putMsg.getContainerId(), deletionTimeMs);
     for (Host host : hosts) {
-      host.addMessage(partitionId, new MessageInfo(id, buffer.remaining(), true), buffer.duplicate());
+      host.addMessage(partitionId,
+          new MessageInfo(id, buffer.remaining(), true, putMsg.getAccountId(), putMsg.getContainerId(), deletionTimeMs),
+          buffer.duplicate());
     }
   }
 
   /**
    * Constructs an entire message with header, blob properties, user metadata and blob content.
    * @param id id for which the message has to be constructed.
-   * @return {@link ByteBuffer} representing the entire message.
+   * @param accountId accountId of the blob
+   * @param containerId containerId of the blob
+   * @return a {@link Pair} of {@link ByteBuffer} and {@link BlobProperties} representing the entire message and its
+   *         Blob properties
    * @throws MessageFormatException
    * @throws IOException
    */
-  private ByteBuffer getPutMessage(StoreKey id) throws MessageFormatException, IOException {
+  private Pair<ByteBuffer, BlobProperties> getPutMessage(StoreKey id, short accountId, short containerId)
```

Comment:
We create blobId outside and so the wish to have the same accountId and containerId. Wrt MessageInfo, I would prefer to go with BlobProperties as its already available where as in case of MessageInfo, we need to create one with properties from BlobProperties. Also, since this is test code, I guess exposing BlobProps should be okay.